src/query/sql/src/planner/optimizer/rule/rewrite/rule_eager_aggregation.rs

```diff
@@ -190,56 +215,181 @@ impl Rule for RuleEagerAggregation {
+                        }
+                    }
+
+                    // TODO(dousir9): add comments in the future.
+                    if let Some(idx) = eval_scalar_items.get(&old_index) && !avg_components.contains_key(&old_index) {
+                        let eval_scalar_item = &mut eval_scalar.items[*idx];
+                        if let ScalarExpr::BoundColumnRef(column) = &mut eval_scalar_item.scalar {
+                            let mut column_binding = &mut column.column;
+                            if column_binding.index == old_index {
+                                column_binding.index = new_index;
+                                if func_name == "count" {
+                                    column_binding.data_type = Box::new(DataType::Nullable(Box::new(
+                                        DataType::Number(NumberDataType::UInt64),
+                                    )));
+                                    eval_scalar_item.scalar = ScalarExpr::CastExpr(CastExpr {
+                                        span: None,
+                                        is_try: false,
+                                        argument: Box::new(eval_scalar_item.scalar.clone()),
+                                        target_type: Box::new(DataType::Number(NumberDataType::UInt64)),
+                                    });
                                 }
                             }
+                            success = true;
                         }
                     }
                 }
 
+                // AVG(C) = SUM(C) / COUNT(NOTNULL C)
+                for (sum_index, count_index) in avg_components.iter() {
+                    if let Some(idx) = eval_scalar_items.get(sum_index) {
+                        let eval_scalar_item = &mut eval_scalar.items[*idx];
+                        eval_scalar_item.scalar = ScalarExpr::FunctionCall(FunctionCall {
+                            span: None,
+                            func_name: "divide".to_string(),
+                            params: vec![],
+                            arguments: vec![
+                                ScalarExpr::BoundColumnRef(BoundColumnRef {
+                                    span: None,
+                                    column: ColumnBinding {
+                                        database_name: None,
+                                        table_name: None,
+                                        column_name: "_eager".to_string(),
+                                        index: old_to_new[sum_index],
+                                        data_type: Box::new(DataType::Number(
+                                            NumberDataType::Float64,
+                                        )),
+                                        visibility: Visibility::Visible,
+                                    },
+                                }),
+                                ScalarExpr::CastExpr(CastExpr {
```

Comment:
`count` is always uint64, do not need a cast.

Reply:
The data type generated by sum is nullable, we need to convert it to non-nullable type, I will add some code comments to it.